alpha/lib/model/FileSync.php

```diff
@@ -43,15 +43,104 @@ public static function createForFileSyncKey ( FileSyncKey $key )
 		return $file_sync;
 	}
 
+	private function generateKey()
+	{
+		return implode("_", array($this->getObjectId(), $this->getObjectType(), $this->getObjectSubType()));
+	}
+
+	public function encrypt()
+	{
+		if (!$this->shouldEncryptFile())
+			return;
+
+		$this->setEncryptionKey($this->generateKey());
+		$this->save();
+
+		$key = $this->getEncryptionKey();
+		$realPath = realpath($this->getFullPath());
+		KalturaLog::debug("Encrypting content of fileSync " . $this->id . ". key is: [$key] in path [$realPath]");
+		$plainData = kFileBase::getFileContent( $realPath);
+		$cryptData = kEncryptFileUtils::encryptData($plainData, $key, $this->getIv());
+		kFileBase::setFileContent( $realPath, $cryptData);
+	}
+
+	public function decrypt()
+	{
+		$realPath = realpath($this->getFullPath());
+		$fileData = kFileBase::getFileContent( $realPath);
+		if (!$this->isEncrypted()) 
+		{
+			KalturaLog::info("File of fileSyncId [$this->id] in path $realPath is not encrypted");
+			return $fileData;
+		}
+
+		$key = $this->getEncryptionKey();
+		KalturaLog::debug("Decrypting content of fileSync " . $this->id . ". key is: [$key]");
+		$plainData = kEncryptFileUtils::decryptData($fileData, $key, $this->getIv());
+		return $plainData;
+	}
+
+	public function shouldEncryptFile()
+	{
```

Comment:
If file size is too big than return false, take the max file size from base.ini.